src/main/java/org/sbml/spatial/segmentation/Inference.java

```diff
@@ -0,0 +1,69 @@
+package org.sbml.spatial.segmentation;
+
+import java.awt.Color;
+import java.awt.image.BufferedImage;
+import java.io.File;
+
+import javax.imageio.ImageIO;
+
+import org.datavec.image.loader.NativeImageLoader;
+import org.deeplearning4j.nn.graph.ComputationGraph;
+import org.nd4j.linalg.api.ndarray.INDArray;
+
+public class Inference {
+	private ComputationGraph model;
+	// private ZooModel model;
+	private String path;
+	private String directory;
+	private static final int WIDTH = 128;
+	private static final int HEIGHT = 128;
+	private static final int CHANNELS = 3;
+
+	public Inference(ComputationGraph model, String path, String directory) {
+		this.model = model;
+		this.path = path;
+		this.directory = directory;
+	}
+
+	/*
+	 * public Inference(ZooModel model, String path) { this.model = model; this.path
+	 * = path; }
+	 */
+
+	// For inferring the model
+	public void imgOut() {
+		try {
+			NativeImageLoader loader = new NativeImageLoader(HEIGHT, WIDTH, CHANNELS);
+			ImageType img = new ImageType(new File(path));
+			BufferedImage bufferedBGR = img.getBGRBufferedImage();
+			INDArray imageNative = loader.asMatrix(bufferedBGR);
+			imageNative.shapeInfoToString();
+			imageNative = imageNative.reshape(1, CHANNELS, HEIGHT, WIDTH);
+			imageNative = imageNative.divi(255f);
+			INDArray[] output = model.output(imageNative);
+			for (INDArray out : output) {
+				out = out.reshape(1, HEIGHT, WIDTH);
+				// out = out.permute(2,1,0);
+				BufferedImage bufferedImage = new BufferedImage(WIDTH, HEIGHT, BufferedImage.TYPE_BYTE_GRAY);
+				for (int i = 0; i < WIDTH; i++) {
+					for (int j = 0; j < HEIGHT; j++) {
+						float f = out.getFloat(new int[] { 0, j, i });
+						int gray = (int) (f * 255.0);
+						if (gray > 127) {
+							bufferedImage.setRGB(i, j, Color.WHITE.getRGB());
+						} else {
+							bufferedImage.setRGB(i, j, Color.BLACK.getRGB());
+						}
+					}
+				}
+
+				ImageIO.write(bufferedImage, "tif", new File(directory + File.separator + "outputUnet.tif"));
```

Comment:
Do we infer 1 image per run? If not this will overwrite previous images

Reply:
Yes, we do infer only one image at a time. So, the problem of overwriting will not occur as it is. But it will probably be better to return a buffered image and not save it directly via the function. I will get working on it.

Reply:
Inferring only one image isn't really practical. One of our objectives is to automatically segment images. So we should take a path as an argument and infer all images (single file or all files in a directory) and output the result.

PS: we can make a new PR for this. For now, just leave it as is and create a new branch/issue for it